### ISSUE TITLE
Gallery 1.5.1rc1 Image deactivation has no effect

### ISSUE THREAD

**Issue by exside** [splittingred/Gallery#189]
I just tried the newest update of Gallery but I had to uninstall btw. restore the 1.4.0 version because in Gallery 1.5.1rc1 I can't deactivate Images in Galleries anymore...if I click "Update Item" on an image the popup opens then I untick the "active" checkbox but after saving and closing the popup the image isn't grayed out and if I click "update item" again, the checkbox is ticked again, so it gets not saved.

**Comment by mgbowman** [comment]
I discovered this same issue yesterday and today I've manage to fix it.

In `core/components/gallery/processors/item/update.class.php : beforeSave()` there's a call to `setCheckbox(true)`

The docs for `setCheckbox( string $k, boolean $force = false )` ...

> Special helper method for handling checkboxes. Only set value if passed or $force is true, and check for a not 
> empty value or string 'false'.

Since Extjs does not send 0 for the unchecked active field, the album item never gets set to 0 (inactive). However, the inverse is true - you have an inactive gallery item (by manually setting it in the db) and you activate it through the manager - everything works as expected.

Solution:

```
--- a/core/components/gallery/processors/mgr/item/update.class.php
+++ b/core/components/gallery/processors/mgr/item/update.class.php
@@ -33,7 +33,7 @@ class GalleryItemUpdateProcessor extends modObjectUpdateProcessor {
     public $languageTopics = array('gallery:default');

     public function beforeSet() {
-        $this->setCheckbox('active');
+        $this->setCheckbox('active', true);
         return parent::beforeSet();
     }

```

Hope to see this integrated in rc2 :)

Cheers!
-mgb

**Comment by splittingred** [comment]
Was fixed in 1.5.0-pl
